### PR TITLE
feat: add MerklePayoutStrategyFactory

### PR DESCRIPTION
**Pull request by hmrtn**
#### Description

- Implement MerklePayoutStrategyFactory
- Add tests + documentation for Factory contract
- Create dummy MerklePayoutStrategyImplementation (the actual functionality can be implemented later)
- Create deploy script for 
  - deploy factory 
  - deploy implementation 
  - link factory to implementation
  - creating a merkle contract 
  
  closes https://github.com/gitcoinco/grants-round/issues/1096

### CODE REVIEW

File: packages/contracts/contracts/payoutStrategy/MerklePayoutStrategyImplementation.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        }
+    }
+
+    function _setClaimed(uint256 _index) private {
+        uint256 claimedWordIndex = _index / 256;
+        uint256 claimedBitIndex = _index % 256;
+        claimedBitMap[claimedWordIndex] |= (1 << claimedBitIndex);
+    }
+
+    function updateDistribution(bytes calldata encodedDistribution) external override onlyRole(ROUND_OPERATOR_ROLE) {
```
Comment (hmrtn) [inline packages/contracts/contracts/payoutStrategy/MerklePayoutStrategyImplementation.sol:70]: Should these methods be restricted to the round operator - set at the init?

File: packages/contracts/contracts/payoutStrategy/MerklePayoutStrategyImplementation.sol [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        token.safeTransfer(_claimee, _amount);
+
+        emit FundsClaimed(_index, _claimee, _amount);
+    }
+
+    function reclaimFunds(IERC20 _token) external onlyRole(ROUND_OPERATOR_ROLE) {
+
+        uint256 _balance = _token.balanceOf(address(this));
+
+        // CHECK: Should this be the sender?
```
Comment (hmrtn) [inline packages/contracts/contracts/payoutStrategy/MerklePayoutStrategyImplementation.sol:112]: Should be sender? (i.e. round operator)

**Comment by DanieleSalatti** [comment]
Added deployment scripts

**Comment by thelostone-mc** [comment]
This PR should be against payout-integration branch and not main!

File: packages/contracts/contracts/payoutStrategy/MerklePayoutStrategyImplementation.sol (hunk 1 above)
Comment (thelostone-mc) [reply packages/contracts/contracts/payoutStrategy/MerklePayoutStrategyImplementation.sol:70]: there is a modifier which can be used at the interface level.
Use payout-integration branch as the starting point

**Review by bhargavaparoksham — APPROVED** [review]

**Pull request merged**